tests/test_update_file_manager.py

```diff
@@ -71,3 +77,792 @@ def test_check_if_update_based_on_last_modified(database, mocker, mock_db):
+TestOrg,192.168.1.2,443-445,udp""",
+            [
+                ("TestOrg", "192.168.1.1", "80/tcp"),
+                ("TestOrg", "192.168.1.2", "443/udp"),
+                ("TestOrg", "192.168.1.2", "444/udp"),
+                ("TestOrg", "192.168.1.2", "445/udp"),
+            ],
+        ),
+        # Testcase2: File with invalid line format.
+        (
+            """Organization,IP,Ports Range,Protocol
+TestOrg,192.168.1.1,80
+TestOrg,192.168.1.2,443-445,udp""",
+            [
+                ("TestOrg", "192.168.1.2", "443/udp"),
+                ("TestOrg", "192.168.1.2", "444/udp"),
+                ("TestOrg", "192.168.1.2", "445/udp"),
+            ],
+        ),
+        # Testcase3: Empty file.
+        (
+            "",
+            [],
+        ),
+    ],
+)
+def test_read_ports_info(mocker, mock_db, tmp_path, test_data, expected_calls):
+    """Test read_ports_info with different file contents."""
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    test_file = tmp_path / "ports_info.csv"
+    test_file.write_text(test_data, encoding="utf-8")
+    mocker.patch("builtins.open", mock_open(read_data=test_data))
+    update_manager.read_ports_info(str(test_file))
+    for call in expected_calls:
+        mock_db.set_organization_of_port.assert_any_call(*call)
+
+
+@pytest.mark.parametrize(
+    "test_data, file_name, expected_db_call",
+    [
+        # Testcase1: Update ports_used_by_specific_orgs.csv.
+        (
+            """Organization,IP,Ports Range,Protocol
+            TestOrg,192.168.1.1,80,tcp
+            TestOrg,192.168.1.2,443-445,udp""",
+            "ports_used_by_specific_orgs.csv",
+            "set_organization_of_port",
+        ),
+        # Testcase2: Update services.csv.
+        (
+            """ssh,22,tcp
+            http,80,tcp""",
+            "services.csv",
+            "set_port_info",
+        ),
+    ],
+)
+def test_update_local_file(
+    mocker, mock_db, tmp_path, test_data, file_name, expected_db_call
+):
+    """Test update_local_file with a valid file."""
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    update_manager.new_hash = "test_hash"
+    test_file = tmp_path / file_name
+    test_file.write_text(test_data, encoding="utf-8")
+    mocker.patch("builtins.open", mock_open(read_data=test_data))
+    result = update_manager.update_local_file(str(test_file))
+    mock_db.set_TI_file_info.assert_called_once_with(
+        str(test_file), {"hash": "test_hash"}
+    )
+    assert result is True
+
+
+def test_check_if_update_online_whitelist_download_updated(mocker, mock_db):
+    """Update period passed, download succeeds."""
+    mock_db.get_TI_file_info.return_value = {"time": 0}
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    update_manager.online_whitelist = "https://example.com/whitelist.txt"
+
+    mock_requests = mocker.patch("requests.get")
+    mock_requests.return_value.status_code = 200
```

Comment:
hey @Sekhar-Kumar-Dash using this 
```py
update_manager.download_file = Mock()
update_manager.download_file.return_value = True
```
instead of 
```py
 mock_requests = mocker.patch("requests.get")
 mock_requests.return_value.status_code = 200
```
is way mor ereadable, because when i was checking this function, i spent a lot of time debugging how come you didn't mock  download_file but yet the tests are passing?, it's because you did mock the "requests.get" right?
so always mock the function call, not the calls inside the function call